### PR TITLE
[WIP] Feature/emergencies

### PR DESCRIPTION
**Pull request by ascalamogna**

### CODE REVIEW

**Comment by dereklieu** [comment]
Map component changes:

1. Moves map init to `utils/get-new-map` (aka setting map drag settings, etc.)
1. Adds an abstract map component to `components/map/map`.
1. Abstract map component takes `geojson, filters, layers` props as a way to drive rendering.
1. Parent map components are still responsible for setting event listeners and opening popups, and get a `setupListeners` hook to do this.

In the abstract map component, difference checking is done using object-level checks aka `thisFilters === nextFilters` so we just need to make sure to create new `filter`, `layer`, and `geoJSON` objects rather than extending current ones, but this is good practice anyway and already happening.

File: app/assets/scripts/components/connected/emergencies-table.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+      return (
+        <Fold title='Latest Alerts'>
+          <p>Oh no! An error ocurred getting the stats.</p>
+        </Fold>
+      );
+    }
+
+    if (!data.objects.length) {
+      return (
+        <Fold title='Latest Alerts'>
```
Comment (danielfdsilva) [inline app/assets/scripts/components/connected/emergencies-table.js]: Here and above you have "Latest Alerts" but below you have "Latest Emergencies". Is this correct?

Comment (dereklieu) [reply app/assets/scripts/components/connected/emergencies-table.js]: 👍 they should all be "Latest Emergencies"

File: app/assets/scripts/reducers/emergencies.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+      countries[c.iso].records.push(record);
+    });
+  });
+
+  const geoJSON = {
+    type: 'FeatureCollection',
+    features: Object.keys(countries).map(iso => {
+      const { country, records } = countries[iso];
+
+      const properties = {
```
[the remark this answers is not in the document]
Comment (dereklieu) [reply app/assets/scripts/reducers/emergencies.js]: 👍

**Comment by dereklieu** [comment]
@danielfdsilva I've incorporated your feedback and synced with `develop`. This should be good to go, pending one last review.

**Pull request merged**